frontend/public/components/utils/dropdown.jsx

```diff
@@ -283,9 +283,22 @@ class Dropdown_ extends DropdownMixin {
   }
 
   applyTextFilter_(autocompleteText, items) {
-    const { autocompleteFilter } = this.props;
+    const { autocompleteFilter, weightedSort } = this.props;
     if (autocompleteFilter && !_.isEmpty(autocompleteText)) {
-      items = _.pickBy(items, (item, key) => autocompleteFilter(autocompleteText, item, key));
+      if (weightedSort) {
+        // Use fuzzysort
+        let totalItems = _.map(items, (item) => item.props);
+        const result = autocompleteFilter(autocompleteText, totalItems);
+        totalItems = {};
+        _.forEach(result, (item) => {
+          const key = `${item.obj.name}-${item.obj.kind}`;
+          totalItems[key] = items[key];
```

Comment:
I'm unsure where this key is coming from. What is setting values using this index in `items`? Will this always be the format?

Technically speaking, this is not unique since you can have the same `kind` and `name` in different API groups.

Reply:
I don't think we need it to be unique :thinking: 
From what I have checked all the places (wherever listdropdwon is used) use this format for the `key` in the `items` object. So, it shouldn't be an issue.